charts/opendj/templates/stateful-set.yaml

```
@@ -0,0 +1,57 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: opendj
+  labels:
+    app: opendj
+    {{- template "opendj.labels" }}
+spec:
+  serviceName: opendj
+  replicas: 1
+  selector:
+    matchLabels:
+      app: opendj
+  template:
+    metadata:
+      labels:
+        app: opendj
+    spec:
+      serviceAccountName: opendj-sa
+      securityContext:
+        # The OpenDJ process runs as a non-root user (1001) in its container. We need to make sure
+        # that the mounted volume is writeable for that user.
+        fsGroup: 1001
+      containers:
+      - name: opendj
+        image: "{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 1389
+          protocol: TCP
+        - containerPort: 1636
+          protocol: TCP
+        - containerPort: 4444
+          protocol: TCP
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        env:
+        - name: BASE_DN
+          # This value should be replaced by environment.
+          value: "dc=dsde-my-environment,dc=broadinstitute,dc=org"
```

```suggestion
          value: "dc=dsde-my-environment,dc=broadinstitute,dc=org"
```
Not familiar enough with this to know the proper syntax here. What are the valid options? If we set it to say `dc=dsde-{{ .Values.global.terraEnv }}` will it only work for current Terra environments?

---

This is one of those values that spans a few places. It's the openDJ base domain name, so all the things that talk to it need to be using the same value. I don't know if the broadinstitute org values are special.

firecloud develop definition: https://github.com/broadinstitute/firecloud-develop/blob/0bd6665585421bb8302799e10adb810e67ebef74/configure.rb#L56

Sam today: https://github.com/broadinstitute/firecloud-develop/blob/dev/base-configs/sam/sam.conf.ctmpl#L32
and every Apache proxy configuration (which we will need in some form but don't have yet for k8s.): https://github.com/broadinstitute/firecloud-develop/search?q=LDAP_BASE_DOMAIN&unscoped_q=LDAP_BASE_DOMAIN

---

but to answer your question, there's nothing special about the 'dsde-env' value AFAIK, we just need to agree on it.

---

OK, I'll make it dsde-[env] so that it's easy to keep it consistent with the current configs